### PR TITLE
Resources/article file connections

### PR DESCRIPTION
**Pull request by BaconFinger**
### Description
Implementing article file connections. There are many thing-to-file connections in Fortnox so some decisions were made under the knowledge that this would be expanded in the future.

### Checklist
- [x] Code compiles correctly
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            Assert.IsNotNull(response);
+            Assert.AreEqual(fileConnections.First().FileId, response.FileId);
+        }
+
+        [TestMethod]
+        public async Task ItCanCreateAConnectionForAFile()
+        {
+            var fileId = await CreateFile();
+            var articles = await GetArticles();
+            var article = !articles.Any() ? await CreateArticle() : articles.First();
```
Comment (GGAlanSmithee) [inline Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs:52]: An alternative:

`articles.FirstOrDefault() ?? await CreateArticle()`

File: Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        {
+            var request = new FortnoxApiRequest(connectionSettings.AccessToken, connectionSettings.ClientSecret);
+            await ArticleFileConnectionService.DeleteArticleFileConnection(request, fileId);
+        }
+
+        private async Task<ArticleFileConnection> CreateConnection()
+        {
+            var fileId = await CreateFile();
+            var articles = await GetArticles();
+            var article = !articles.Any() ? await CreateArticle() : articles.First();
```
Comment (GGAlanSmithee) [inline Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs:121]: as above

**Comment by GGAlanSmithee** [comment]
@BaconFinger looking good to me, some minor comments

File: Fortnox.NET.Tests/Fortnox.NET.Tests/Fortnox.NET.Tests.csproj [hunk 3]
```diff
@@ -15,6 +15,7 @@
         <PackageReference Include="MSTest.TestAdapter" Version="1.4.0" />
         <PackageReference Include="MSTest.TestFramework" Version="1.4.0" />
         <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+        <PackageReference Include="System.Drawing.Common" Version="4.7.0" />
```
[the remark this answers is not in the document]
Comment (BaconFinger) [reply Fortnox.NET.Tests/Fortnox.NET.Tests/Fortnox.NET.Tests.csproj:18]: Generating an image file to attach to an article.

File: Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs (hunk 1 above)
Comment (BaconFinger) [reply Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs]: Done

File: Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs (hunk 2 above)
Comment (BaconFinger) [reply Fortnox.NET.Tests/Fortnox.NET.Tests/ArticleFileConnectionsTest.cs]: Done

File: Fortnox.NET.Tests/Fortnox.NET.Tests/Fortnox.NET.Tests.csproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (GGAlanSmithee) [reply Fortnox.NET.Tests/Fortnox.NET.Tests/Fortnox.NET.Tests.csproj:18]: Does it have to be an image, or could we do something [simpler](https://docs.microsoft.com/en-us/dotnet/api/system.io.file?view=netcore-3.1#examples)?

[the remark this answers is not in the document]
Comment (BaconFinger) [reply Fortnox.NET.Tests/Fortnox.NET.Tests/Fortnox.NET.Tests.csproj:18]: It has to be an image. I've tried with a text file and it didn't take.

[the remark this answers is not in the document]
Comment (GGAlanSmithee) [reply Fortnox.NET.Tests/Fortnox.NET.Tests/Fortnox.NET.Tests.csproj:18]: Alright

**Comment by GGAlanSmithee** [comment]
Looks good to me! @brokenprogrammer ?

**Pull request merged**